src/Response.php

```diff
@@ -137,7 +137,7 @@ public function withStatus($code, $reasonPhrase = ''): self
 
         $new = clone $this;
         $new->statusCode = (int) $code;
-        if ('' == $reasonPhrase && isset(self::$phrases[$new->statusCode])) {
+        if ('' === $reasonPhrase && isset(self::$phrases[$new->statusCode])) {
```

Comment:
Nothing protects from passing a null reason phrase here which would bypass this condition